### PR TITLE
[VariationBundle]  Form registration, missing specs and CS 

### PR DESCRIPTION
**Pull request by aRn0D**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | refactoring 
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | no
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
:+1: :+1: :+1:

**Comment by aRn0D** [comment]
Taxonomy is coming :v:

**Comment by nicolasricci** [comment]
@aRn0D I have the following error after updating this PR

```
PHP Fatal error:  Call to undefined method Symfony\Component\Config\Definition\Builder\ScalarNodeDefinition::scalarNode() in /vendor/sylius/sylius/src/Sylius/Bundle/VariationBundle/DependencyInjection/Configuration.php on line 125
```

**Comment by aRn0D** [comment]
@nicolasricci I will have a look this evenening

File: src/Sylius/Bundle/VariationBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -113,7 +118,13 @@ private function addClassesSection(ArrayNodeDefinition $node)
                                     ->scalarNode('model')->isRequired()->end()
                                     ->scalarNode('controller')->defaultValue('Sylius\Bundle\ResourceBundle\Controller\ResourceController')->end()
                                     ->scalarNode('repository')->cannotBeEmpty()->end()
-                                    ->scalarNode('form')->defaultValue('Sylius\Bundle\VariationBundle\Form\Type\OptionType')->end()
+                                    ->arrayNode('form')
+                                        ->addDefaultsIfNotSet()
+                                        ->children()
+                                            ->scalarNode('default')->defaultValue('Sylius\Bundle\VariationBundle\Form\Type\OptionType')
+                                            ->scalarNode('choice')->defaultValue('Sylius\Bundle\ResourceBundle\Form\Type\ResourceChoiceType')
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/VariationBundle/DependencyInjection/Configuration.php]: Missing ``` ->end() ```

File: src/Sylius/Bundle/VariationBundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -113,7 +118,13 @@ private function addClassesSection(ArrayNodeDefinition $node)
                                     ->scalarNode('model')->isRequired()->end()
                                     ->scalarNode('controller')->defaultValue('Sylius\Bundle\ResourceBundle\Controller\ResourceController')->end()
                                     ->scalarNode('repository')->cannotBeEmpty()->end()
-                                    ->scalarNode('form')->defaultValue('Sylius\Bundle\VariationBundle\Form\Type\OptionType')->end()
+                                    ->arrayNode('form')
+                                        ->addDefaultsIfNotSet()
+                                        ->children()
+                                            ->scalarNode('default')->defaultValue('Sylius\Bundle\VariationBundle\Form\Type\OptionType')
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/VariationBundle/DependencyInjection/Configuration.php]: Missing ``` ->end() ```

**Comment by michalmarcinkowski** [comment]
@nicolasricci I think I found solution (see above).

**Comment by nicolasricci** [comment]
@michalmarcinkowski indeed well spotted :+1:

**Comment by aRn0D** [comment]
I will fix it in the user PR

**Comment by aRn0D** [comment]
@michalmarcinkowski thanks !

**Comment by nicolasricci** [comment]
You are also missing the default key in sylius.yml
https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/CoreBundle/Resources/config/app/sylius.yml#L103

**Comment by aRn0D** [comment]
pushed in #2824

**Pull request merged**